lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -40,8 +40,18 @@ def import_module(name):  # type: ignore[misc]
     reload_module = reload  # type: ignore[name-defined]  # pylint:disable=undefined-variable
 
 try:
-    from importlib.abc import TraversableResources
+    try:
+        # Available on Python >= 3.11
+        # We ignore the import error that will trigger when running mypy with
+        # older Python versions.
+        from importlib.resources.abc import TraversableResources  # type: ignore[import]
+    except ImportError:
+        # Used with Python 3.9 and 3.10 only
+        # This member is still available as an alias up until Python 3.14 but
+        # is deprecated as of Python 3.12.
+        from importlib.abc import TraversableResources
```

Comment:
Hey, this is also a perfect use for our shiny new declarative "hey, we can remove this code when we no longer support Python X.Y" lint rule (added in #81071). 

```suggestion
        from importlib.abc import TraversableResources  # deprecated: description='TraversableResources move' python_version='3.10'
```